examples/src/main/java/com/mapbox/navigation/examples/core/MapMatchingActivity.kt

```diff
@@ -0,0 +1,396 @@
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.matching.v5.MapboxMapMatching
+import com.mapbox.api.matching.v5.models.MapMatchingResponse
+import com.mapbox.geojson.Point
+import com.mapbox.mapboxsdk.camera.CameraUpdateFactory
+import com.mapbox.mapboxsdk.location.LocationComponent
+import com.mapbox.mapboxsdk.location.LocationComponentActivationOptions
+import com.mapbox.mapboxsdk.location.modes.CameraMode
+import com.mapbox.mapboxsdk.location.modes.RenderMode
+import com.mapbox.mapboxsdk.maps.MapboxMap
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback
+import com.mapbox.mapboxsdk.maps.Style
+import com.mapbox.navigation.base.internal.accounts.UrlSkuTokenProvider
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.reroute.RerouteController
+import com.mapbox.navigation.core.reroute.RerouteState
+import com.mapbox.navigation.core.trip.session.TripSessionState
+import com.mapbox.navigation.core.trip.session.TripSessionState.STARTED
+import com.mapbox.navigation.core.trip.session.TripSessionState.STOPPED
+import com.mapbox.navigation.core.trip.session.TripSessionStateObserver
+import com.mapbox.navigation.examples.R
+import com.mapbox.navigation.examples.utils.Utils
+import com.mapbox.navigation.examples.utils.Utils.PRIMARY_ROUTE_BUNDLE_KEY
+import com.mapbox.navigation.examples.utils.Utils.getRouteFromBundle
+import com.mapbox.navigation.examples.utils.extensions.toPoint
+import com.mapbox.navigation.ui.camera.DynamicCamera
+import com.mapbox.navigation.ui.camera.NavigationCamera
+import com.mapbox.navigation.ui.map.NavigationMapboxMap
+import com.mapbox.navigation.ui.map.NavigationMapboxMapInstanceState
+import com.mapbox.navigation.utils.internal.ifNonNull
+import java.lang.ref.WeakReference
+import kotlinx.android.synthetic.main.activity_basic_navigation_layout.*
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import timber.log.Timber
+
+/**
+ * This activity shows how to fetch DirectionsRoute with MapMatching
+ *
+ */
+@SuppressLint("MissingPermission")
+class MapMatchingActivity : AppCompatActivity(), OnMapReadyCallback {
+
+    private companion object {
+        private const val MAP_INSTANCE_STATE_KEY = "navigation_mapbox_map_instance_state"
+        private const val INTERVAL_MILLISECONDS = 1000L
+        private const val FASTEST_INTERVAL_MILLISECONDS = 500L
+    }
+
+    private var mapboxNavigation: MapboxNavigation? = null
+    private var navigationMapboxMap: NavigationMapboxMap? = null
+    private var mapInstanceState: NavigationMapboxMapInstanceState? = null
+    private var directionsRoute: DirectionsRoute? = null
+    private lateinit var localLocationEngine: LocationEngine
+    private val locationEngineCallback = MyLocationEngineCallback(this)
+    private var destination: Point? = null
+    private var locationComponent: LocationComponent? = null
+    private val urlSkuTokenProvider = TestUrlSkuTokenProvider()
```

Comment:
This is not needed right? Actually for the BYOR use case SKU shouldn't be provided as requests are not done through `MapboxNavigation` / the SDK.

Reply:
removed